### ISSUE TITLE
Pricing Feedback

### ISSUE THREAD

**Issue by rangav** [rangav/thunder-client-support#1177]
This is continuation of previous [thread](https://github.com/rangav/thunder-client-support/issues/1170)

## Subscription Plans

The extension was free for the **last 2 years**, We are launching subscription plans for businesses to support our development costs.

- The core extension will be free for personal/individual use.
- Businesses have until `August 18th, 2023` to make a decision about subscription.
- The pricing is affordable when compared to Insomnia or Postman.

## Why subscription plan is needed?
- A subscription plan cost is for `extension software + team sharing features + updates` to be used by businesses. 
- The revenue generated from subscriptions will be useful to cover below costs
  - Development Costs
  - Team Salaries
  - Operational Costs
  - Future development costs
  - Support costs
  - Sales Costs
  

### Pricing

- **Free** - Use of extension for personal/individual use
- **Startup Plan** - $5 per user/month billed `monthly`  or $3 per user/month billed `yearly`
-  **Business Plan** - $10 per user/month billed `monthly`  or $7 per user/month billed `yearly`
-  **Enterprise Plan** - $15 per user/month billed `monthly`  or $12 per user/month billed `yearly`
- Please visit [thunderclient](https://www.thunderclient.com/pricing) website for more details

Please let us know your feedback about pricing below

**Comment by rangav** [comment]
One clarification - We are charging users for use of `Extension Software + Team Share features` for commercial purpose.

We are not charging just for saving to data to git project.

**Comment by sanderkooger** [comment]
Does the data get stored in the same place as the current plug-in?

**Comment by leoek** [comment]
Hmm, this is quite an unfortunate turn of events since we started using it in one of our projects and liked the simplicity. Maybe this is a fortunate timing though before we are using it more.

Some of my thoughts:

1. There is no hosting costs involved, because everything is done locally in my own repositories. Charging for a completely local, client side feature seems weird.
2. Pricing is just too high. Postman offers many more features for 12$ (More than I actually need, but some which would be a nice addtion here.) Even if thunder would offer the same feature set, it wouldn't be worth the accounting overhead for this small difference.

**Comment by leoek** [comment]
> One clarification - We are charging users for use of Extension Software + Team Share features for commercial purpose.

This seems worse than the original wording. What are these Team share features and Extension Software I'd be paying for?

**Comment by bfdog** [comment]
Is it possible to use the free version from a business if the Team Share features are not used?

**Comment by rangav** [comment]
> Is it possible to use the free version from a business if the Team Share features are not used?

Yes you can use free plan for business purpose also

**Comment by rangav** [comment]
> > One clarification - We are charging users for use of Extension Software + Team Share features for commercial purpose.
> 
> This seems worse than the original wording. What are these Team share features and Extension Software I'd be paying for?

Hi @leoek the subscription cost is to use all the features of extension and sharing features 

All features listed here
https://github.com/rangav/thunder-client-support

Sharing features 
https://github.com/rangav/thunder-client-support#team

what wording would you suggest

**Comment by rangav** [comment]
> Hmm, this is quite an unfortunate turn of events since I tried it for a personal project and we also as a team started using it in one of our projects and liked the simplicity. Maybe this is a fortunate timing though before using it more.
> 
> Some of my thoughts:
> 
> 1. There is no hosting costs involved, because everything is done locally in my own repositories. Charging for a completely local, client side feature seems weird.
> 2. Even for personal use I want my collection to be part of the git repository. I wouldn't to mantain and backup the collections seperately.
> 3. Pricing is just too high. Postman offers many more features for 12$ (More than I actually need, but some which would be a nice addtion here.) Even if thunder would offer the same feature set, it wouldn't be worth the accounting overhead for this small difference.
> 4. For that price I would at least expect support and continued updates, however the license states:
>    > The Licensor has no obligation to provide updates or upgrades to the Software, or otherwise maintain or support the Software.

We still have cloud cost for the website, but the main costs are mentioned below 
- Development Costs
- Team Salaries
- Operational Costs
- Future development costs
- Support costs
- Sales Costs

will review about the support wording and get back to you.

**Comment by LuisEngelniederhammer** [comment]
> One clarification - We are charging users for use of `Extension Software + Team Share features + updates` for commercial purpose.
> 
> We are not charging just for saving to data to git project.

Could you clarify what's ment by `updates`?

So does that mean if I use your extension on an enterprise/commercial project, save my request collections etc. within the git repository and keep that within github, but will not take/receive any updates, we would not have to pay for a subscription?

Thank you.

**Comment by rangav** [comment]
@LuisEngelniederhammer the subscription cost also includes future updates.

**Comment by DuboisCecile** [comment]
Hi,

Does that mean that we won't be able to use Git sync, even for personal projects ? I am currently using it to save my tests in my own repositories, so I can use them on different computers (logging with my github account). And sometimes, I also need to share some code with friends, not for commercial use (when I need help for debugging, for example).

You wrote : 

> One clarification - We are charging users for use of Extension Software + Team Share features for commercial purpose.

But how can we still use Team Share features for personal uses ?

Best regards

**Comment by DuboisCecile** [comment]
> Hi @DuboisCecile for this use case we kept `startup plan` affordable who want use sharing features, suitable for individual or small teams to use it for personal/business purpose.
> 
> Or You can use free extension software without sharing features.

Ok, so you shouldn't write "for commercial purpose", because it is also for personal use...

**Comment by sanderkooger** [comment]
Hey @DuboisCecile, 

The functionally @rangav is trying to monetize is the ability to store your information within a location you prefer, therefore breaking the current functionality. 

He will charge for: 
- The ability to store per project in the project repo on your hard drive
- the ability to store information per project (everything will be in the same files outside of git for free)
- the ability to access the information already stored inside your existing repo if you do not have a subscription (taking your collections hostage) 

Basically, you get charged for usage of your own storage, and a git push command. Mind you that @rangav keeps censoring my comments, because he knows I'm right, but he wants to word it differently. 

Previous discussions are here in a discussion he closed after no one agreed with him for monetizing in this way, Now he's trying again with a reworded version of the same thing. 

https://github.com/rangav/thunder-client-support/issues/1170

@rangav As all users including me are telling you, We want to support you, We would like to pay, 

Unfortunately, your inability to listen to that, and poor business decisions, are making that really hard. Don't censor those people, as that is not abuse. You can choose to report this comment as abuse if you want, but that is not going to help, quite the opposite.

**Comment by rangav** [comment]
Hey @sanderkooger all your further comments will be deleted for abusive behavior. If you are not happy with subscription plans feel free to any other service that suits your needs

**Comment by rangav** [comment]
> > Hi @DuboisCecile for this use case we kept `startup plan` affordable who want use sharing features, suitable for individual or small teams to use it for personal/business purpose.
> > Or You can use free extension software without sharing features.
> 
> Ok, so you shouldn't write "for commercial purpose", because it is the same for personal use...

Thanks for feedback, will update it.

**Comment by sanderkooger** [comment]
@rangav just the mere fact that you need to delete on point comments is pretty abusive from your side. It also proves my point. 

Good luck, on this path, It's destroying all you have built.

**Comment by cgatian** [comment]
Unfortunately, this means we will be migrating off this plugin in favor of the already org-wide usage of Postman.

**Comment by rangav** [comment]
Hi @cgatian I can understand that, Unfortunately, we have to generate revenue to support dev costs and team salaries. Otherwise, the project will be difficult to support and provide future updates.

Is it related to pricing cost or are you expecting it to be free?

**Comment by rangav** [comment]
Hi All,

I can think of an alternative way for a subscription plan ( not based on sharing features).

Here is my proposal.

**Fee Plan** 
- Limited Git features
- 1 Collection, 
- 1 Environment
- 25 requests can be added to the collection.

**Subscription Plan**
- Unlimited Collection, 
- Unlimited Environments
- Unlimited requests can be added to the collection.

I think either way both are similar to me, But let me know your feedback, If the above looks better to you?

**Comment by vasansr** [comment]
@rangav I totally support the move for monetization, but you guys are going in the same direction as Postman: charging for team features. GitHub and Bitbucket went with different monetization models. One charged for non-public repositories and the other charged for collaboration. That made both of them viable choices for beginners, who then, being accustomed to one or the other stuck with it when their needs grew and they needed a paid plan.

IMO, charging for collaboration in TC will make it no different from Postman, which is widely more popular at the moment and the defacto choice of many. Just like @cgatian people will prefer a paid subscription to Postman over a paid subscription to TC. My prediction (I hope I am wrong because I love TC) is that you will find it hard to survive head-on against Postman.

So I suggest we look for alternative premium features that we can charge for. But the idea is small teams and individuals won't have to pay, but larger companies will. Some ideas:

* Volume-based: Rate limiting, number of requests per collection, number of collections etc.
* Feature-based: One or more of: documentation / CI-CD integration / running all requests in a collection

**Comment by cgatian** [comment]
> Is it related to pricing cost or are you expecting it to be free?

I am expecting it to be free. Yes, that's how the plugin has functioned for the last **two years**. Now that it's moving to a paid model, I can no longer use this at my organization, even though I've been a vocal supporter of it. I understand we all want to get paid for our time, but it will make it impossible for us to continue its use (it's not the leader in this space by any means). 

Furthermore, the pricing model move has soured my use of this plugin, and I won't be recommending it in the future because I feel like I've been duped into using it and advocating it and then blindsided with a paywall. Only to be reminded that I needed to read the license carefully, which states this move was planned from the start. The **one sentence** in the license file should've been a little clearer in the intentions, perhaps maybe even something on the README that stated:

> **"We plan on making money with this plugin one day, only once we receive enough feedback and features and get enough people to integrate it within their projects, probably once we cross 2 million downloads. Until then, use as much as you want and contribute PRs. Thanks."**

Ultimately, you can do what you want with your plugin. I hope the pricing model pays off and you can receive the compensation you're hoping for.

**Comment by rangav** [comment]
Thanks @cgatian for the feedback, Some user might be not be happy with the launch of pricing plans.

As I said earlier we still have to start charging users to generate some revenue to pay our bills.

Keeping all features free is not a financially viable option for us.

**Comment by rangav** [comment]
> Hi All,
> 
> I can think of an alternative way for a subscription plan ( not based on sharing features).
> 
> Here is my proposal.
> 
> **Fee Plan**
> 
> * Limited Git features
> * 1 Collection,
> * 1 Environment
> * 25 requests can be added to the collection.
> 
> **Subscription Plan**
> 
> * Unlimited Collection,
> * Unlimited Environments
> * Unlimited requests can be added to the collection.
> 
> I think either way both are similar to me, But let me know your feedback, If the above looks better to you?

Hi All, Please leave your feedback on the above. Based on the feedback, we can change to feature-based subscription plans rather than sharing features.

**Comment by ingmferrer** [comment]
I don't plan on telling you how to monetize your product, I'm a little disappointed that the core features that are currently free will no longer be. I think gitlens pro relies on creating new features on subscription only, but everything else that was free before was still free.

Anyway, good luck!

**Comment by leoek** [comment]
> > > One clarification - We are charging users for use of Extension Software + Team Share features for commercial purpose.
> > 
> > 
> > This seems worse than the original wording. What are these Team share features and Extension Software I'd be paying for?
> 
> Hi @leoek the subscription cost is to use all the features of extension and sharing features
> 
> All features listed here https://github.com/rangav/thunder-client-support
> 
> Sharing features https://github.com/rangav/thunder-client-support#team
> 
> what wording would you suggest?

If I am paying for a vscode extension which runs completely client side, I would at least expect some kind of a support plan included and continous updates. Currently the license states this:

> The Licensor has no obligation to provide updates or upgrades to the Software, or otherwise maintain or support the Software.

I am not buying into software from a "company" which excludes Liability, Warranty and the right to alter the software. This is fine for an oss project, because I'll still have the escape hatch of fixing something myself or not oss free software, because it must be seen as a nice to have which can't be relied upon. Sorry but this is just an impractical combination.

**Comment by rangav** [comment]
Hi @leoek I have removed that sentence now from the license. We will provide regular updates and fixes to software which have purchased a subscription.

Let me know if any other wording is an issue.

**Comment by leoek** [comment]
This is a stp into the right direction, however the statement still holds: With a subscription service I'd expect some feature or service which warrants a subscription. With it being just a local extension a license bought with a one time payment seems better suitable.

Example: MS office includes 1tb cloud storage if I pay the subscription. If I just want to open some office files every once in a while I can buy the license without subscription.

**Comment by rangav** [comment]
Software that charge for one time fees does not provides any future updates. 

Our subscription plan includes future updates too.

**Comment by bdharrington7** [comment]
First, thank you for building this plugin. It filled some needs I was looking for that Postman lacked (e.g. a client right in VSCode). I completely understand the need to make money from your work! But I'm worried that you may be going about it the wrong way.

The proposed subscription model is a non-starter for me any many others, because:
* In the technical sense: It requires internet access just to use a feature that doesn't inherently need internet access.
* In the ethical sense: it's charging users to store data where they want, on their own system, using their own resources. 

I don't know if the git-sync feature is what you determined to be the most-depended on feature (or how), but are there not other features in the pipeline that you could monetize? Someone else also mentioned GitLens' subscription feature as a possible (more developer friendly) monetization model.

I also feel like it's kind of an affront to long-time users that actually helped you shape TC into what it is today. Filing bug / feature reports is not a zero-cost effort.

I am also at a loss for a feasible monetization model for plugins, so I empathize with what you're trying to do. You do have to do _something_ to make a living, but it's not crystal clear what that is. Unfortunately, the proposed subscription / fee model would prevent me from continuing to use this plugin. 

So, some feedback: I think the fee model would be more feasible in my case, if it didn't require an internet connection, and it wasn't so restrictive on the number of calls. I would be ok with not getting future updates (beyond bug and security fixes to a point), and would be willing to pay a fee for a new major version with compelling features.

**Comment by rangav** [comment]
@bdharrington7 one time fee based model will not for extensions as they need to get updates and bug fixes.

We have 2 options for subscription 
- Sharing based model as currently used at the top
- Or feature based model.

Internet connection is only needed at the time of signing to activate subscription, we can think of workaround for this - this is not a big deal.

**Comment by leoek** [comment]
> Software that charge for one time fees does not provides any future updates.
> 
> Our subscription plan includes future updates too.
> 
> We don't have any cloud service, for that reason we reduced the plan costs compared to insomnia or postman.
> 
> But cloud costs only a fraction of overall costs. All the other costs still apply for any company, see my first comment at the top

This means I'd be paying for potential features which I probably don't need. Maintenance updates are usually included even with one time purchases.

**Comment by rangav** [comment]
@leoek unfortunately extensions will work this way, we can't send separate updates for different users.

Standalone software can work without updates.

@leoek you also need to understand no matter what way we charge there always be some users will be unhappy 

and each user will have a different opinion and feedback. 

It will be difficult for us to come up with a paid plan that makes everyone happy.

**Comment by leoek** [comment]
@rangav I believe my point cam across. Just take this as a feedback to improve your business plan.

I am a little sad that this will probably mean the end of my usage:

- For personal projects Postman is free (I can commit the collection to git there too) and there is no additional feature here to adjustify a subscription for me.
- For our team project test candidate another subscription is hard to justify and a complete switch not worth.

**Comment by gitblit** [comment]
@rangav My shop is on Postman, of which I am not a huge fan... but it gets the job done. TC is way more fun & straight-forward to use. It's also nice for screensharing, demos, and tech talks since you don't have to leave VS Code. It's a great tool and you should be proud.

Maybe this is a terrible idea, but have you considered leaving the extension alone and trying to monetize the CLI test runner (and juicing it up for Github/other integration)?

In my org, we export Postman collections, commit them to git, and run Newman in Github Actions on every PR and/or on post-merge regressions (depends on repo). IDK what a reasonable pricing model around that might look like, but there is a 3rd option you could explore.

**Comment by fleetimee** [comment]
> @rangav My shop is on Postman, of which I am not a huge fan... but it gets the job done. TC is way more fun & straight-forward to use. It's also nice for screensharing, demos, and tech talks since you don't have to leave VS Code. It's a great tool and you should be proud.
> 
> Maybe this is a terrible idea, but have you considered leaving the extension alone and trying to monetize the CLI test runner (and juicing it up for Github/other integration)?
> 
> In my org, we export Postman collections, commit them to git, and run Newman in Github Actions on every PR and/or on post-merge regressions (depends on repo). IDK what a reasonable pricing model around that might look like, but there is a 3rd option you could explore.

Actually postman recently released vscode extension 

so i guess to to say goodbye to tc 

![](https://github.com/rangav/thunder-client-support/assets/45744788/d1465f13-b167-44b5-9daa-90f9edce1877)

**Comment by rangav** [comment]
Thanks everyone, all your feedback is noted.
